agent/unit_state.go

```diff
@@ -71,12 +73,19 @@ type UnitStatePublisher struct {
 // them to the Registry every 5s. Heartbeat objects are also published as they
 // are received on the channel.
 func (p *UnitStatePublisher) Run(beatchan <-chan *unit.UnitStateHeartbeat, stop chan bool) {
+	var publishPeriod time.Duration
+	if p.ttl > 30*time.Second {
```

Comment:
realistically we don't have a 10mins ttl, it was just an example. Back then, having the publishPeriod being an arithmetic function of the ttl looked like a better solution than having it being a geometric function of the ttl. In the future, a better solution might be more in the lines of `publishPeriod = ttl - (etcdtimeout*2)`